website/ember_osf_web/decorators.py

```diff
@@ -15,7 +15,10 @@ def ember_flag_is_active(flag_name):
     def decorator(func):
         @functools.wraps(func)
         def wrapped(*args, **kwargs):
-            request.user = _get_current_user()
+            # Waffle does not enjoy NoneTypes as user values.
+            mock_user = type('Mock User', (object,), {'is_authenticated': False})()
```

Comment:
This looks like a possible memory leak--a new class is defined for every request that hits this. Why not just a single class?

```python
class MockUser(object):
    is_authenticated = False

request.user = _get_current_user() or MockUser()
```